models/POP/model_mod.f90

```diff
@@ -356,9 +338,10 @@ subroutine static_init_model()
 call init_interp()
 
 !> @todo 'pop.r.nc' is hardcoded in dart_pop_mod.f90
-domain_id = add_domain('pop.r.nc', nfields, &
-                       var_names = variable_table(1:nfields, VAR_NAME_INDEX), &
-                       update_list = update_var_list(1:nfields))
+! parse_variables converts the character table that was read in from
+! model_nml:model_state_variables and returns a state_var_type that
+! can be passed to add_domain
```

Comment:
This model_mod is missing the logic to use the default state variables if no model_state_variables = ' '
before calling add_domain

https://github.com/NCAR/DART/blob/f2deaa27499fea386d4b72d44a4134f4adf2ab62/models/POP/model_mod.f90#L2774-L2778

```
if model_state_variables = ' '
  call use_default_state_variables( state_variables )
  'blah blah using default state variables'
endif 
domain_id = add_domain('pop.r.nc', parse_variables(model_state_variables))